extension/chrome/elements/compose-modules/compose-my-pubkey-module.ts

```diff
@@ -8,11 +8,13 @@ import { Lang } from '../../../js/common/lang.js';
 import { Ui } from '../../../js/common/browser/ui.js';
 import { ViewModule } from '../../../js/common/view-module.js';
 import { ComposeView } from '../compose.js';
-import { ContactStore } from '../../../js/common/platform/store/contact-store.js';
+import { Str } from '../../../js/common/core/common.js';
+import { OpenPGPKey } from '../../../js/common/core/crypto/pgp/openpgp-key.js';
 
 export class ComposeMyPubkeyModule extends ViewModule<ComposeView> {
 
   private toggledManually = false;
+  private wkdLongids: { [acctEmail: string]: string[] } = {};
```

Comment:
It is preferable to compare fingerprints over longids whenever possible - fingerprints are slightly more secure, and there is no advantage using longids.

The only place that makes sense to use longids is when looking up public keys of a signed message, because the signature only contains a longid. Similarly encrypted message only mentions longid of the key it's encrypted for. In all other situations, let's use fingerprints.

Reply:
Done, but I had to parse the `KeyInfo.private` field as we don't currently have the primary fingerprint there, only `longid` and a set of all `fingerprints`. Or do we have to do a "must include all fingerprints" check? Also, do we need to attach our key only if "Sign" option is ticked?